p2p/peer.py

```diff
@@ -693,6 +641,17 @@ def handle_sub_proto_msg(self, cmd: protocol.Command, msg: protocol._DecodedMsgT
             if actual_td > self.head_td:
                 self.head_hash = actual_head
                 self.head_td = actual_td
+
+        if cmd_type in self.pending_requests:
+            request, future = self.pending_requests[cmd_type]
+            try:
+                request.validate_response(msg)
+            except ValidationError:
+                pass
+            else:
+                future.set_result(msg)
+                self.pending_requests.pop(cmd_type)
+
```

Comment:
We could move the whole block above into `BasePeer.handle_sub_proto_msg()` and then avoid having to duplicate it in both `ETHPeer` and `LESPeer`, no?